From 5420a3c4bd70a487ffe413713e1ee5c9ce3501ef Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Fri, 20 Mar 2020 11:41:44 +0200
Subject: [PATCH] [SW-10750] goya2: max ppboot image

Define max allowed ppboot image to load so we don't
exceed the allowed time to reach recovery in case both images
fail

Change-Id: I7dc32e5febdb7e1873c841ca047058505ae5fd82
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 8b71fc6..44465fa 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -222,6 +222,10 @@
 	flash_ppboot_pcie_offset = (uint32_t)(FLASH_64KB_PADD +
 					      flash_image_offset);
 	ppboot_pcie_size = (uint16_t)spi_flash_readl(flash_ppboot_pcie_offset);
+	if (ppboot_pcie_size > SECUREBOOT_MAX_PPBOOT_IMG_SIZE) {
+		arcboot_rom_status_update_bringup(image_progress, PROGRESS_BRINGUP_TOO_BIG);
+		return ARCBOOT_BRINGUP_ERROR;
+	}
 
 	/* update in PSOC_SECURITY the current img fw_version */
 	flash_image_fw_ver = spi_flash_readl(flash_image_offset +
diff --git a/common/secureboot.h b/common/secureboot.h
index f3fc9e9..a69bb63 100644
--- a/common/secureboot.h
+++ b/common/secureboot.h
@@ -13,6 +13,7 @@
 #define IMG_LEN_SIZE			4
 #define FW_VERSION_SIZE			4
 #define CRC32_SIZE			4
+#define SECUREBOOT_MAX_PPBOOT_IMG_SIZE	46080
 
 #define SECUREBOOT_HASH_BLOCK_CALC_BUF	512
 
diff --git a/include/arcboot.h b/include/arcboot.h
index 564f4e6..8ac067d 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -74,6 +74,7 @@
 	PROGRESS_BRINGUP_START,
 	PROGRESS_BRINGUP_DMA,
 	PROGRESS_BRINGUP_SECURITY,
+	PROGRESS_BRINGUP_TOO_BIG,
 };
 
 enum arcboot_rom_progress {
@@ -90,19 +91,21 @@
 	PROGRESS_BRINGUP_PCI_PRI,		/* 10 */
 	PROGRESS_BRINGUP_PCI_PRI_DMA,
 	PROGRESS_BRINGUP_PCI_PRI_SECURITY,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG,
 	PROGRESS_BRINGUP_PCI_SEC,
-	PROGRESS_BRINGUP_PCI_SEC_DMA,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY,	/* 15 */
+	PROGRESS_BRINGUP_PCI_SEC_DMA,		/* 15 */
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG,
 	PROGRESS_PCIE_IN_RECOVERY,
 	PROGRESS_PPBOOT_START,
-	PROGRESS_PPBOOT_FINISH,
+	PROGRESS_PPBOOT_FINISH,			/* 20 */
 	PROGRESS_BRINGUP_RAM_PRI,
-	PROGRESS_BRINGUP_RAM_PRI_DMA,		/* 20 */
+	PROGRESS_BRINGUP_RAM_PRI_DMA,
 	PROGRESS_BRINGUP_RAM_PRI_SECURITY,
 	PROGRESS_BRINGUP_RAM_SEC,
-	PROGRESS_BRINGUP_RAM_SEC_DMA,
+	PROGRESS_BRINGUP_RAM_SEC_DMA,		/* 25 */
 	PROGRESS_BRINGUP_RAM_SEC_SECURITY,
-	PROGRESS_ARCBOOT_RAM_START		/* 25 */
+	PROGRESS_ARCBOOT_RAM_START
 };
 
 enum security_location {
